components/shared/Select.tsx

```diff
@@ -0,0 +1,88 @@
+import { Fragment, useState, FC, SetStateAction } from 'react'
+import { Listbox, Transition } from '@headlessui/react'
+import { CheckIcon, ChevronUpDownIcon } from '@heroicons/react/20/solid'
+import { twMerge } from 'tailwind-merge'
+import classnames from 'classnames'
+
+interface SelectProps {
+  options: { id: number; name: string }[]
+  defaultSelected?: { id: number; name: string }
+  onSelect: (value: { id: number; name: string }) => void
+}
+
+export const tw: typeof classnames = (...params) =>
```

Comment:
you'll be using this utility function eeeeeverywhere, I recommend you create a `tw.ts` file. For example, you could've used it for the BaseButton component since it uses classnames conditionally